tests/unit/customizations/history/test_show.py

```diff
@@ -0,0 +1,813 @@
+        }
+        self.formatter.display(event)
+        collected_output = ensure_text_type(self.output.getvalue())
+        self.assertIn(
+            'with body: ' + self.get_pretty_xml(xml_body),
+            collected_output
+        )
+
+    def test_display_http_request_with_xml_payload_and_whitespace(self):
+        xml_body = '<?xml version="1.0" ?><foo><bar>text</bar></foo>'
+        event = {
+            'event_type': 'HTTP_REQUEST',
+            'id': 'my-id',
+            'request_id': 'some-id',
+            'payload': {
+                'method': 'GET',
+                'url': 'https://myservice.us-west-2.amazonaws.com',
+                'headers': {},
+                'body': self.get_pretty_xml(xml_body)
+            },
+            'timestamp': 0,
+        }
+        self.formatter.display(event)
+        collected_output = ensure_text_type(self.output.getvalue())
+        # The XML should not be prettified more than once if the body
+        # of the request was already prettied.
+        self.assertIn(
+            'with body: ' + self.get_pretty_xml(xml_body),
+            collected_output
+        )
+
+    def test_display_http_request_with_json_struct_payload(self):
+        event = {
+            'event_type': 'HTTP_REQUEST',
+            'id': 'my-id',
+            'request_id': 'some-id',
+            'payload': {
+                'method': 'GET',
+                'url': 'https://myservice.us-west-2.amazonaws.com',
+                'headers': {},
+                'body': '{"foo": "bar"}'
+            },
+            'timestamp': 0,
+        }
+        self.formatter.display(event)
+        collected_output = ensure_text_type(self.output.getvalue())
+        self.assertIn(
+            ('with body: {\n'
+             '    "foo": "bar"\n'
+             '}'),
+            collected_output
+        )
+
+    def test_shares_api_number_across_events_of_same_api_call(self):
+        api_event = {
+            'event_type': 'API_CALL',
+            'id': 'my-id',
+            'request_id': 'some-id',
+            'payload': {
+                'service': 'ec2',
+                'operation': 'DescribeRegions',
+                'params': {}
+            },
+            'timestamp': 0,
+        }
+        http_event = {
+            'event_type': 'HTTP_REQUEST',
+            'id': 'my-id',
+            'request_id': 'some-id',
+            'payload': {
+                'method': 'GET',
+                'url': 'https://myservice.us-west-2.amazonaws.com',
+                'headers': {},
+                'body': 'This is my body'
+            },
+            'timestamp': 0,
+        }
+        self.formatter.display(api_event)
+        self.formatter.display(http_event)
+        collected_output = ensure_text_type(self.output.getvalue())
+        self.assertIn('[0] API call made', collected_output)
+        self.assertIn('[0] HTTP request sent', collected_output)
+
+    def test_display_http_response(self):
+        event = {
```

Comment:
These display tests all seem to have the same format.  You could cut down on duplication by having a custom assertion method:

```python
self.assert_output(
    for_event={
    },
    contains_output='...'
)
```